src/ParallelAlgorithms/Interpolation/InterpolationTargetDetail.hpp

```diff
@@ -522,13 +521,34 @@ auto block_logical_coords(const db::DataBox<DbTags>& box,
   }
 
   // Time-independent case.
-  return ::block_logical_coordinates(
-      domain, InterpolationTargetTag::compute_target_points::points(
-                  box, tmpl::type_<Metavariables>{}, temporal_id));
+  return ::block_logical_coordinates(domain, coords);
+}
+
+/// Version of block_logical_coords that computes the interpolation
+/// points on the fly.  This version is called when the interpolation
+/// points are (or might be) time-dependent in the
+/// InterpolationTargetTag's frame.
+///
+/// This version of block_logical_coordinates is called when there
+/// is an Interpolator ParallelComponent.
+///
+/// Currently one Action directly calls this version of block_logical_coords:
+/// - SendPointsToInterpolator (called by AddTemporalIdsToInterpolationTarget
+///                             and by FindApparentHorizon)
+template <typename InterpolationTargetTag, typename DbTags,
+          typename Metavariables, typename TemporalId>
+auto block_logical_coords(const db::DataBox<DbTags>& box,
+                          Parallel::GlobalCache<Metavariables>& cache,
```

Comment:
I'd expect the `cache` can be constant. It only needs to be non-const if being used for communication